src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int DEFAULT_SIZE = 0;
+    private Object[] elements;
+    private int capacity;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+        size = DEFAULT_SIZE;
+        capacity = DEFAULT_CAPACITY;
+    }
+
+    private void resize() {
+        capacity = (int) (capacity * 1.5);
+        Object[] elementsResized = new Object[capacity];
+        System.arraycopy(elements, 0, elementsResized, 0, elements.length);
```

Comment:
The checklist requires avoiding utility classes like `Arrays`. Instead of `Arrays.copyOf`, use `System.arraycopy()` to copy elements. However, in this line, `System.arraycopy()` is already used, so ensure that `Arrays.copyOf` is not used elsewhere in the code.